### ISSUE TITLE
YouTube --write-info-json only writes info on available automatic subtitles if they have been downloaded

### ISSUE THREAD

**Issue by peteruhrig** [ytdl-org/youtube-dl#20520]
---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.03.18*. If it's not, read [this FAQ entry](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2019.03.18**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/ytdl-org/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/ytdl-org/youtube-dl#faq) and [BUGS](https://github.com/ytdl-org/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### Output:

```
kermit@frog:/yt$ youtube-dl --skip-download -o "%(id)s.mp4" --write-info-json --all-subs -v https://www.youtube.com/watch?v=bdkXTKZNlSo
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--skip-download', '-o', '%(id)s.mp4', '--write-info-json', '--all-subs', '-v', 'https://www.youtube.com/watch?v=bdkXTKZNlSo']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.03.18
[debug] Python version 3.5.2 (CPython) - Linux-4.4.0-138-generic-x86_64-with-Ubuntu-16.04-xenial
[debug] exe versions: ffmpeg 2.8.15, ffprobe 2.8.15
[debug] Proxy map: {}
[youtube] bdkXTKZNlSo: Downloading webpage
[youtube] bdkXTKZNlSo: Downloading video info webpage
WARNING: video doesn't have subtitles
[debug] Default format spec: bestvideo+bestaudio/best
[info] Writing video description metadata as JSON to: bdkXTKZNlSo.info.json
...
<end of log>
```

### Section `automatic_captions` is empty in info.json

With the command given above, the section `automatic_captions` remains empty in the info.json file (see [attachment](https://github.com/ytdl-org/youtube-dl/files/3028171/bdkXTKZNlSo.info.json.zip)). It will be filled if I run the following command:

```
kermit@frog:/yt$ youtube-dl --skip-download -o "%(id)s.mp4" --write-info-json --all-subs --write-auto-sub -v https://www.youtube.com/watch?v=bdkXTKZNlSo
```

However, this will download 104 subtitle files I do not want.
My plan is to download the JSON file first and then decide on the basis of available subtitles and video resolutions whether I want to download the video and which video format to request. But it appears I cannot do this at the moment.
In short: If --all-subs is switched on, I would expect the `automatic_captions` section to be populated in the info.json.

Thanks and best wishes,
Peter

**Comment by jgalp** [comment]
Is this a valid issue and is anyone working on it? I am a student at the University of Michigan looking to contribute to an open source project for my upper level software engineering class.
